packages/invoices/src/api.ts

```diff
@@ -245,14 +250,27 @@ export const listQueryFn = async <T extends ObjectWithStringKeys>(queryParams: {
 };
 
 export const detailQueryFn = (id: string) => async (): Promise<Invoice> => {
-  const result = await request(Environment.API_URL, getDetailQuery(), {
-    id,
-  });
-  const { invoice } = result;
+  const result = await api.invoice({ id });
 
-  const mapped = { ...invoice, lines: invoice.lines.nodes };
+  const { invoice } = result;
 
-  return mapped;
+  if (invoice.__typename === 'InvoiceNode') {
+    return {
+      ...invoice,
+      lines:
+        invoice.lines.__typename === 'InvoiceLineConnector'
+          ? invoice.lines.nodes
+          : [],
+      pricing: {
+        totalAfterTax:
+          invoice.pricing.__typename === 'InvoicePricingNode'
+            ? invoice.pricing.totalAfterTax
+            : 0,
+      },
+    };
+  } else {
+    throw new Error('uhoh');
```

Comment:
So uhh..

We're typed, but this is.. not great.

I am just not sure about this conversion of GQL types into 'normal types'  to use without the app. I am quite sure it is a good idea, but I am hesistent as I don't want to make things confusing.

I believe here we can simple throw an error if there is an error when getting the invoice price or the lines or if there is an error with the invoice itself. In all of those cases it is surely something wrong with the invoice, in which case we just have to show a 'Nope'!.

But, we can likely do something with the description. Maybe..

Reply:
needs to be done somewhere - better to have it in one place and have the rest of the app work with normal data